### PR TITLE
Switched to pytest asserts for issue #776

### PR DESCRIPTION
**Pull request by bishnucit**
For issue #776 Switched to pytest asserts.

### CODE REVIEW

**Comment by davehunt** [comment]
FWIW @bishnucit and I have already talked about our preference to work on feature branches rather than submitting pull requests from master. Unfortunately he had some issues with his local repository and ended up making all of these changes via GitHub's web UI. This is a critical patch and we must take care not to change intentions of the asserts, so I would like any changes to be tested locally and not to rely entirely on Travis / Jenkins results.

That all said, there is a flake8 error that Travis has detected:

> ./tests/mobile/test_home.py:52:1: W293 blank line contains whitespace

Please address this, and my other comments once I've completed the initial review.

File: pages/desktop/regions/island.py [hunk 1]
```diff
@@ -26,7 +25,7 @@ def pager(self):
         try:
             return self.Pager(self.testsetup, self._root.find_element(*self._pager_locator))
         except NoSuchElementException:
-            Assert.fail('Paginator is not available')
+            assert not 'Paginator is not available'
```
Comment (davehunt) [inline pages/desktop/regions/island.py]: The closest equivalent to `Assert.fail` is `raise AssertionError(msg)`

File: pages/desktop/themes.py [hunk 2]
```diff
@@ -102,7 +101,9 @@ class ThemesDetail(Base):
     def is_the_current_page(self):
         # This overrides the method in the Page super class.
         actual_page_title = self.page_title
-        Assert.not_none(re.match(self._page_title_regex, actual_page_title), 'Expected the current page to be the themes detail page.\n Actual title: %s' % actual_page_title)
+
+        assert re.match(self._page_title_regex, actual_page_title) is not None, 'Expected the current page to be the themes detail page.\n Actual title: %s' % actual_page_title
+        assert re.match(self._page_title_regex, actual_page_title) is not None, 'Expected the current page to be the themes detail page.\n Actual title: %s' % actual_page_title
```
Comment (davehunt) [inline pages/desktop/themes.py]: Please remove the duplicate line and the empty line change.

File: pages/desktop/themes.py [hunk 3]
```diff
@@ -126,8 +127,8 @@ class ThemesBrowse(Base):
     @property
     def is_the_current_page(self):
         # This overrides the method in the Page super class.
-        Assert.true(self.is_element_present(*self._themes_grid_locator),
-                    'Expected the current page to be the themes browse page.')
+        assert self.is_element_present(*self._themes_grid_locator), 'Expected the current page to be the themes browse page.'
+        assert self.is_element_present(*self._themes_grid_locator), 'Expected the current page to be the themes browse page.'
```
Comment (davehunt) [inline pages/desktop/themes.py]: Remove the duplicate line, and please restore the line break.

File: pages/desktop/themes.py [hunk 4]
```diff
@@ -145,5 +146,6 @@ def sort_by(self):
 class ThemesSearchResultList(SearchResultList):
     _results_locator = (By.CSS_SELECTOR, 'ul.personas-grid div.persona-small')
 
-    class ThemesSearchResultItem(SearchResultList.SearchResultItem):
-        _name_locator = (By.CSS_SELECTOR, 'h6 > a')
+
+class ThemesSearchResultItem(SearchResultList.SearchResultItem):
+    _name_locator = (By.CSS_SELECTOR, 'h6 > a')
```
Comment (davehunt) [inline pages/desktop/themes.py]: This is an unrelated change, please revert it.

File: tests/desktop/test_complete_themes.py [hunk 5]
```diff
@@ -21,17 +19,19 @@ def test_that_complete_themes_can_be_sorted_by_name(self, mozwebqa):
         complete_themes_page.click_sort_by("name")
         addons = complete_themes_page.addon_names
         addons_set = set(addons)
-        Assert.equal(len(addons), len(addons_set), "There are duplicates in the names")
+        assert len(addons) == len(addons_set), 'There are duplicates in the names'
         addons_orig = addons
         addons.sort()
-        [Assert.equal(addons_orig[i], addons[i]) for i in xrange(len(addons))]
+        for i in xrange(len(addons)):
+            assert (addons_orig[i] == addons[i])
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: No need for the brackets here.

File: tests/desktop/test_complete_themes.py [hunk 6]
```diff
@@ -21,17 +19,19 @@ def test_that_complete_themes_can_be_sorted_by_name(self, mozwebqa):
...
         complete_themes_page.paginator.click_next_page()
         addons = complete_themes_page.addon_names
         addons_set = set(addons)
-        Assert.equal(len(addons), len(addons_set), "There are duplicates in the names")
+        assert len(addons) == len(addons_set), 'There are duplicates in the names'
         addons_orig = addons
         addons.sort()
-        [Assert.equal(addons_orig[i], addons[i]) for i in xrange(len(addons))]
+        for i in xrange(len(addons)):
+            assert (addons_orig[i] == addons[i])
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: No need for the brackets here.

File: tests/desktop/test_complete_themes.py [hunk 7]
```diff
@@ -41,12 +41,12 @@ def test_that_complete_themes_can_be_sorted_by_updated_date(self, mozwebqa):
         complete_themes_page.click_sort_by("recently updated")
         addons = complete_themes_page.addon_names
         addons_set = set(addons)
-        Assert.equal(len(addons), len(addons_set), "There are duplicates in the names")
+        assert len(addons) == len(addons_set), 'There are duplicates in the names'
         updated_dates = complete_themes_page.addon_updated_dates
-        Assert.is_sorted_descending(updated_dates)
+        assert updated_dates.sort(reversed=True)
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: This is not doing what you expect. We need to compare `updated_dates` to `updated_dates.sort(reversed=True)`

File: tests/desktop/test_complete_themes.py [hunk 8]
```diff
@@ -41,12 +41,12 @@ def test_that_complete_themes_can_be_sorted_by_updated_date(self, mozwebqa):
...
         complete_themes_page.paginator.click_next_page()
         updated_dates.extend(complete_themes_page.addon_updated_dates)
-        Assert.is_sorted_descending(updated_dates)
+        assert updated_dates.sort(reversed=True)
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: As above, this is a false positive.

File: tests/desktop/test_complete_themes.py [hunk 9]
```diff
@@ -56,12 +56,12 @@ def test_that_complete_themes_can_be_sorted_by_created_date(self, mozwebqa):
         complete_themes_page.click_sort_by("newest")
         addons = complete_themes_page.addon_names
         addons_set = set(addons)
-        Assert.equal(len(addons), len(addons_set), "There are duplicates in the names")
+        assert len(addons) == len(addons_set), 'There are duplicates in the names'
         created_dates = complete_themes_page.addon_created_dates
-        Assert.is_sorted_descending(created_dates)
+        assert created_dates.sort(reverse=True)
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: As above, this is a false positive.

File: tests/desktop/test_complete_themes.py [hunk 10]
```diff
@@ -56,12 +56,12 @@ def test_that_complete_themes_can_be_sorted_by_created_date(self, mozwebqa):
...
         complete_themes_page.paginator.click_next_page()
         created_dates.extend(complete_themes_page.addon_created_dates)
-        Assert.is_sorted_descending(created_dates)
+        assert created_dates.sort(reverse=True)
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: As above, this is a false positive.

File: tests/desktop/test_complete_themes.py [hunk 11]
```diff
@@ -71,12 +71,12 @@ def test_that_complete_themes_can_be_sorted_by_popularity(self, mozwebqa):
         complete_themes_page.click_sort_by("weekly downloads")
         addons = complete_themes_page.addon_names
         addons_set = set(addons)
-        Assert.equal(len(addons), len(addons_set), "There are duplicates in the names")
+        assert len(addons) == len(addons_set), 'There are duplicates in the names'
         downloads = complete_themes_page.addon_download_number
-        Assert.is_sorted_descending(downloads)
+        assert downloads.sort(reverse=True)
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: As above, this is a false positive.

File: tests/desktop/test_complete_themes.py [hunk 12]
```diff
@@ -71,12 +71,12 @@ def test_that_complete_themes_can_be_sorted_by_popularity(self, mozwebqa):
...
         complete_themes_page.paginator.click_next_page()
         downloads.extend(complete_themes_page.addon_download_number)
-        Assert.is_sorted_descending(downloads)
+        assert downloads.sort(reverse=True)
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: As above, this is a false positive.

File: tests/desktop/test_complete_themes.py [hunk 13]
```diff
@@ -129,7 +129,8 @@ def test_complete_themes_subcategory_page_breadcrumb(self, mozwebqa):
         amo_category_page = complete_themes_page.click_on_first_category()
         expected_breadcrumbs = ['Add-ons for Firefox', 'Complete Themes', selected_category]
 
-        [Assert.equal(expected_breadcrumbs[i], amo_category_page.breadcrumbs[i].text) for i in range(len(amo_category_page.breadcrumbs))]
+        for i in range(len(amo_category_page.breadcrumbs)):
+            assert (expected_breadcrumbs[i] == amo_category_page.breadcrumbs[i].text)
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: No need for the brackets here.

File: tests/desktop/test_complete_themes.py [hunk 14]
```diff
@@ -178,11 +179,10 @@ def test_the_displayed_message_for_incompatible_complete_themes(self, mozwebqa):
...
         for complete_theme in complete_themes:
             if complete_theme.is_incompatible:
-                Assert.true(complete_theme.is_incompatible_flag_visible)
-                Assert.contains('Not available',
-                                complete_theme.not_available_flag_text)
+                assert complete_theme.is_incompatible_flag_visible
+                assert 'Not available' in complete_theme.not_available_flag_text
             else:
-                Assert.false(complete_theme.is_incompatible_flag_visible)
+                assert complete_theme.is_incompatible_flag_visible is False
```
Comment (davehunt) [inline tests/desktop/test_complete_themes.py]: This should be `assert not complete_theme.is_incompatible_flag_visible`

File: tests/desktop/test_details_page.py [hunk 15]
```diff
@@ -21,76 +19,76 @@ class TestDetails:
...
-        Assert.true(details_page.header.is_login_link_visible, "Login links is not visible")
+        assert details_page.header.is_register_link_visible, "Register link is not visible"
+        assert details_page.header.is_login_link_visible, "Login links is not visible"
 
     @pytest.mark.action_chains
     @pytest.mark.nondestructive
     def test_that_dropdown_menu_is_present_after_click_on_other_apps(self, mozwebqa):
         details_page = Details(mozwebqa, "Firebug")
-        Assert.equal(details_page.header.menu_name, "Other Applications")
+        assert details_page.header.menu_name, "Other Applications"
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: False positive, need `==`

File: tests/desktop/test_details_page.py [hunk 16]
```diff
@@ -21,76 +19,76 @@ class TestDetails:
...
         details_page.header.hover_over_other_apps_menu()
-        Assert.true(details_page.header.is_other_apps_dropdown_menu_visible)
+        assert details_page.header.is_other_apps_dropdown_menu_visible
 
     @pytest.mark.nondestructive
     def test_that_addon_name_is_displayed(self, mozwebqa):
         details_page = Details(mozwebqa, "Firebug")
         # check that the name is not empty
-        Assert.not_none(details_page.title, "")
+        assert details_page.title, "" is not None
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: This should be: `assert details_page.title is not None` - ignore the weird empty message argument.

File: tests/desktop/test_details_page.py [hunk 17]
```diff
@@ -21,76 +19,76 @@ class TestDetails:
...
         # check that the summary is not empty
-        Assert.not_none(re.match('(\w+\s*){3,}', details_page.summary))
+        assert re.match('(\w+\s*){3,}', details_page.summary) is not None
 
     @pytest.mark.nondestructive
     def test_that_about_this_addon_is_displayed(self, mozwebqa):
         details_page = Details(mozwebqa, "Firebug")
-        Assert.equal(details_page.about_addon, "About this Add-on")
-        Assert.not_none(re.match('(\w+\s*){3,}', details_page.description))
+        assert details_page.about_addon, "About this Add-on"
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: False positive, needs `==`

File: tests/desktop/test_details_page.py [hunk 18]
```diff
@@ -144,9 +142,9 @@ def test_open_close_functionality_for_image_viewer(self, mozwebqa):
         detail_page = Details(mozwebqa, 'firebug')
 
         image_viewer = detail_page.previewer.click_image()
-        Assert.true(image_viewer.is_visible)
+        assert image_viewer.is_visible
         image_viewer.close()
-        Assert.false(image_viewer.is_visible)
+        assert image_viewer.is_visible is False
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: This should be `assert not image_viewer.is_visible`

File: tests/desktop/test_details_page.py [hunk 19]
```diff
@@ -168,41 +166,41 @@ def test_navigation_buttons_for_image_viewer(self, mozwebqa):
...
-            Assert.equal(image_viewer.image_link.split('/')[-1], image_link[i].split('/')[-1])
+            assert image_viewer.caption == images_title[i]
+            assert image_viewer.image_link.split('/')[-1] == image_link[i].split('/')[-1]
 
             if not i == 0:
-                Assert.true(image_viewer.is_previous_present)
+                assert image_viewer.is_previous_present
             else:
-                Assert.false(image_viewer.is_previous_present)
+                assert image_viewer.is_previous_present is False
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: Should be `assert not image_viewer.is_previous_present`

File: tests/desktop/test_details_page.py [hunk 20]
```diff
@@ -168,41 +166,41 @@ def test_navigation_buttons_for_image_viewer(self, mozwebqa):
...
 
             if not i == image_viewer.images_count - 1:
-                Assert.true(image_viewer.is_next_present)
+                assert image_viewer.is_next_present
                 image_viewer.click_next()
             else:
-                Assert.false(image_viewer.is_next_present)
+                assert image_viewer.is_next_present is False
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: Should be `assert not image_viewer.is_next_present`

File: tests/desktop/test_details_page.py [hunk 21]
```diff
@@ -168,41 +166,41 @@ def test_navigation_buttons_for_image_viewer(self, mozwebqa):
...
-            Assert.equal(image_viewer.image_link.split('/')[-1], image_link[i].split('/')[-1])
+            assert image_viewer.caption == images_title[i]
+            assert image_viewer.image_link.split('/')[-1] == image_link[i].split('/')[-1]
 
             if not i == image_viewer.images_count - 1:
-                Assert.true(image_viewer.is_next_present)
+                assert image_viewer.is_next_present
             else:
-                Assert.false(image_viewer.is_next_present)
+                assert image_viewer.is_next_present is False
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: Should be `assert not...`

File: tests/desktop/test_details_page.py [hunk 22]
```diff
@@ -168,41 +166,41 @@ def test_navigation_buttons_for_image_viewer(self, mozwebqa):
...
 
             if not i == 0:
-                Assert.true(image_viewer.is_previous_present)
+                assert image_viewer.is_previous_present
                 image_viewer.click_previous()
             else:
-                Assert.false(image_viewer.is_previous_present)
+                assert image_viewer.is_previous_present is False
```
Comment (davehunt) [inline tests/desktop/test_details_page.py]: Should be `assert not...`

File: tests/desktop/test_details_page_against_xml.py [hunk 23]
```diff
@@ -8,7 +8,6 @@
 from urllib2 import urlparse
```
Comment (davehunt) [inline tests/desktop/test_details_page_against_xml.py]: Setting myself a bookmark, this is how far I've reached in the review. :smile:

**Comment by rbillings** [comment]
@davehunt - when you have a chance please continue

**Comment by davehunt** [comment]
Apologies for the delay here. Unfortunately this pull request has become out of date, and continuing to review it as it is could risk missing something important. @bishnucit if you don't mind, please rebase and apply changes based on my comments. I will then review it again promptly to avoid it rotting again. If you do not think you have time to work on this, I would be happy with taking your changes as they are and continuing the work myself.

**Comment by davehunt** [comment]
Ah, we also have #779. Please indicate if that replaces this pull request. If so, we can close this one.

**Comment by davehunt** [comment]
Your effort is appreciated here @bishnucit, however as we haven't heard from you in a while I've gone ahead and opened #782 to address this issue. Hopefully you can find another issue to work on.
